### ISSUE TITLE
Training LibriTTS from scratch

### ISSUE THREAD

**Issue by yaguanghu** [k2-fsa/ZipVoice#36]
Hello, I'm trying to train zipvoice model on LibriTTS from scratch. The only difference from the script you provided in tutorial is that I only have 4 GPUs, and I changed the batch size (aka max duration) from 250 to 800. I have trained this model for 21 epochs(50k steps), but when I test it I still get a cacophony. 
The loss of train & valid seems normal, so do you think there is anything wrong with my experiment?
Tks a lot~ 😊  
![Image](https://github.com/user-attachments/assets/810def96-1bfd-4250-aef0-281f0bb491a4)

**Comment by zhu-han** [comment]
I think you are using `--world-size 4`, `--max-duration 200`,  right? Your total batch size is `800s`, while our batch size is `8 * 250s = 2000s`. If you can increase `--max-duration to 500`, I think you can reproduce our results. Otherwise, you may need to decrease the `--base-lr` for this small batch size setting. And 20 epoch is a small number, our model is trained with 60 epochs.

**Comment by yaguanghu** [comment]
Sorry, I may have misunderstood lhotse's batch size calculation. I used --max-duration 800 with 4 gpus, according to your statement the actual batch size is `800 * 4 = 3200s`, right?
By the way, how does your model perform at 20 epochs, is there a reasonable result on LibriTTS/Seed-TTS testset?

**Comment by zhu-han** [comment]
Can you use the following command to see the statistics of the training set?
`lhotse cut describe data/fbank/libritts_cuts_train-all-shuf.jsonl.gz`

**Comment by yaguanghu** [comment]
lhotse cut describe fbank/libritts_cuts_train-all-shuf.jsonl.gz
```
Cut statistics:
╒═══════════════════════════╤═══════════╕
│ Cuts count:               │ 354692    │
├───────────────────────────┼───────────┤
│ Total duration (hh:mm:ss) │ 552:34:49 │
├───────────────────────────┼───────────┤
│ mean                      │ 5.6       │
├───────────────────────────┼───────────┤
│ std                       │ 4.5       │
├───────────────────────────┼───────────┤
│ min                       │ 0.1       │
├───────────────────────────┼───────────┤
│ 25%                       │ 2.3       │
├───────────────────────────┼───────────┤
│ 50%                       │ 4.3       │
├───────────────────────────┼───────────┤
│ 75%                       │ 7.6       │
├───────────────────────────┼───────────┤
│ 99%                       │ 20.9      │
├───────────────────────────┼───────────┤
│ 99.5%                     │ 23.1      │
├───────────────────────────┼───────────┤
│ 99.9%                     │ 27.4      │
├───────────────────────────┼───────────┤
│ max                       │ 43.8      │
├───────────────────────────┼───────────┤
│ Recordings available:     │ 354692    │
├───────────────────────────┼───────────┤
│ Features available:       │ 354692    │
├───────────────────────────┼───────────┤
│ Supervisions available:   │ 354692    │
╘═══════════════════════════╧═══════════╛
SUPERVISION custom fields:
- orig_text (in 354692 cuts)
- snr (in 354692 cuts)
Speech duration statistics:
╒══════════════════════════════╤═══════════╤══════════════════════╕
│ Total speech duration        │ 552:34:49 │ 100.00% of recording │
├──────────────────────────────┼───────────┼──────────────────────┤
│ Total speaking time duration │ 552:34:49 │ 100.00% of recording │
├──────────────────────────────┼───────────┼──────────────────────┤
│ Total silence duration       │ 00:00:00  │ 0.00% of recording   │
╘══════════════════════════════╧═══════════╧══════════════════════╛
```

**Comment by yaguanghu** [comment]
Here is my training log:
[log-train-2025-07-01-08-32-48-0.txt](https://github.com/user-attachments/files/21031587/log-train-2025-07-01-08-32-48-0.txt)

**Comment by Respaired** [comment]
on 30 epochs you'll get a very reasonable sounding speech. but you may have to increase the steps during inference to something around 128 in my experience. the more you train the better it sounds and you can use much smaller steps later on.

**Comment by yaguanghu** [comment]
> on 30 epochs you'll get a very reasonable sounding speech. but you may have to increase the steps during inference to something around 128 in my experience. the more you train the better it sounds and you can use much smaller steps later on.

Thanks for your advice. I've tried to increase the steps to 128, but still got no sound at all or near-noise audio. 
Would you be willing to share your infer script?

**Comment by yaguanghu** [comment]
> > Here is my training log: [log-train-2025-07-01-08-32-48-0.txt](https://github.com/user-attachments/files/21031587/log-train-2025-07-01-08-32-48-0.txt)
> 
> It's very strange. I'll try if I can reproduce this later.

I think I have found the reason why the inference cannot produce reasonable audios. The LibriTTS infer example in the README misses two parameters resulted in the use of phonemes as modeling units. I will make a PR to fix this issue shortly.
But the batch size problem is still wired, and I hope you can help me figure out where the problem is.
Tks a lot again~
